src/types/float.rs

```diff
@@ -140,6 +140,123 @@ impl<'py> FromPyObject<'py> for f32 {
     }
 }
 
+macro_rules! impl_partial_eq_for_float {
+    ($float_type: ty) => {
+        impl PartialEq<$float_type> for Bound<'_, PyFloat> {
+            #[inline]
+            fn eq(&self, other: &$float_type) -> bool {
+                if let Ok(val) = self.extract::<$float_type>() {
+                    val == *other
+                } else {
+                    false
+                }
```

Comment:
We could also write this with `match`:

```suggestion
                match self.extract::<$float_type>() {
                    Ok(val) => val == *other,
                    Err(_) => false,
                }
```

I slightly prefer this, but it doesn't matter much.

Reply:
We can probably use the `PyFloatMethods::value()` and skip the branch entirely. Since we have `Bound<'_, PyFloat>` we are guaranteed that the extraction succeeds.

Reply:
I believe we should stick with `extract()` since we need to take care of comparison between `f32` and `c_double`. The `extract` API properly downcast the underlying `c_double` in `PyFloat` to `f32` and thereby the comparison becomes valid. I tried locally to use the `value()` API , and I notice that when comparing f32 to c_double, the value to be compared becomes `2.019999` vs `2.02`, which results in wrong comparison

Reply:
Hmm, the `f32` extractor for `PyFloat` is literally to extract an `f64` and then cast to `f32`.
https://github.com/PyO3/pyo3/blob/d598d1f4209cfce7be2208f6ca88b7aaceaf6600/src/types/float.rs#L132-L141

So using `value()` and then casting to `f32` should give the same result...

I'll give it a look later

Reply:
If you do something like the following (and the equivalent for the others) the tests seem to pass fine for me

```rust
impl PartialEq<$float_type> for Bound<'_, PyFloat> {
    #[inline]
    fn eq(&self, other: &$float_type) -> bool {
        self.value() as $float_type == *other
    }
}
```

Reply:
So sorry I didn't check thoroughly. Indeed your proposed way works. I've revised the code to use value API instead. Many thanks for correcting me !

Reply:
No worries, glad we agreed 👍